crates/core/src/subscription/module_subscription_actor.rs

```diff
@@ -4,142 +4,46 @@ use super::{
-    AddSubscriber {
-        sender: ClientConnectionSender,
-        subscription: Subscribe,
-    },
-    RemoveSubscriber {
-        client_id: ClientActorId,
-    },
-    BroadcastCommitEvent {
-        event: ModuleEvent,
-        // channel for signaling that all subscriptions have been evaluated.
-        feedback_tx: oneshot::Sender<()>,
-    },
-}
-
-#[derive(Clone, Debug)]
-pub struct ModuleSubscriptionManager {
-    tx: mpsc::UnboundedSender<Command>,
-}
-
-impl ModuleSubscriptionManager {
-    pub fn spawn(relational_db: Arc<RelationalDB>, owner_identity: Identity) -> Self {
-        let (tx, mut rx) = mpsc::unbounded_channel();
-        tokio::spawn(async move {
-            let mut actor = ModuleSubscriptionActor::new(relational_db, owner_identity);
-            while let Some(command) = rx.recv().await {
-                if let Err(e) = actor.handle_message(command).await {
-                    log::error!("error occurred in ModuleSubscriptionActor: {e}")
-                }
-            }
-        });
-        Self { tx }
-    }
-
-    pub fn add_subscriber(&self, sender: ClientConnectionSender, subscription: Subscribe) -> Result<(), NoSuchModule> {
-        self.tx
-            .send(Command::AddSubscriber { sender, subscription })
-            .map_err(|_| NoSuchModule)
-    }
-
-    pub fn remove_subscriber(&self, client_id: ClientActorId) -> Result<(), NoSuchModule> {
-        self.tx
-            .send(Command::RemoveSubscriber { client_id })
-            .map_err(|_| NoSuchModule)
-    }
-
-    pub async fn broadcast_event(&self, client: Option<&ClientConnectionSender>, mut event: ModuleEvent) {
-        match event.status {
-            EventStatus::Committed(_) => {
-                let (tx, rx) = oneshot::channel();
-                self.tx
-                    .send(Command::BroadcastCommitEvent { event, feedback_tx: tx })
-                    .expect("subscription actor panicked");
-                if (rx.await).is_err() {
-                    log::error!("failed to receive acknowledgement of successful evaluation from subscription actor")
-                }
-            }
-            EventStatus::Failed(_) => {
-                if let Some(client) = client {
-                    let message = TransactionUpdateMessage {
-                        event: &mut event,
-                        database_update: Default::default(),
-                    };
-                    let _ = client.send_message(message).await;
-                } else {
-                    log::trace!("Reducer failed but there is no client to send the failure to!")
-                }
-            }
-            EventStatus::OutOfEnergy => {} // ?
-        }
-    }
-
-    pub fn broadcast_event_blocking(&self, client: Option<&ClientConnectionSender>, event: ModuleEvent) {
-        tokio::runtime::Handle::current().block_on(self.broadcast_event(client, event))
-    }
-}
-
-struct ModuleSubscriptionActor {
+pub struct ModuleSubscriptions {
```

Comment:
`Vec<Subscription>>` field of this should be wrap around RwLock, so that we can do concurrent `add_subscription` and `add_subscription` with broadcast.
Current approach of wrapping `ModuleSubscription` will not really allow that.